components/FormContainer.vue

```diff
@@ -0,0 +1,271 @@
+<template>
+	<div>
+		<article class="flex relative flex-col">
+			<div class="grid content-center w-1/2 mx-auto">
+				<Transition>
+					<div v-frag>
+						<component :is="currentState.component" :introduction="currentState.introduction"
+							:value="currentState.value" :followup="currentState.followup" :options="currentState.options"
+							@update="currentState.updateFunction" :textarea="currentState.textarea" :key="stateIndex" />
+
+						<div class="left">
+							<div v-if="stateIndex === 0">
+								<div>
+									<button @click="nextStep()">Start</button>
+									<span class="text-xs">press <strong>Enter</strong> ↵</span>
+									<p class="text-xs mt-2">🕒 Takes 4 minutes</p>
+								</div>
+							</div>
+
+							<button v-if="hasPreviousState" @click="previousStep">Previous</button>
+							<button v-if="hasNextState && stateIndex > 0" @click="nextStep">
+								Next
+							</button>
+							<button v-if="!hasNextState" @click="submitForm">Submit</button>
+						</div>
+					</div>
+				</Transition>
+			</div>
+			<div class="flex flex-wrap p-4">
+				<ProgressBar :percentage="percentage" class="mx-2 mb-2 " />
+			</div>
+		</article>
+	</div>
+</template>
+<script>
+import { mapState } from "vuex";
+
+import FormIntro from "./FormIntro.vue";
+import FormSingleField from "./FormSingleField.vue"
+import FormMultipleSelect from "./FormMultipleSelect.vue"
+import FormMultipleInput from "./FormMultipleInput.vue"
+
+import ProgressBar from "./ProgressBar.vue";
+
+export default {
+	mounted: function () {
+		window.addEventListener('keyup', (e) => { this.keyboardControl(e) })
+	},
+	data: () => {
+		return {
+			stateIndex: 0,
+			submitting: false,
+			formData: {
+				username: "",
+				learnings: [],
+				sharings: [],
+				newLearnings: [],
+				newSharings: [],
+				comment: ''
+			},
+		}
+	},
+	computed: {
+		...mapState(['supabase']),
+		currentLearning: function () {
+			// Get topics that have a value in the learning mapping 
+			// const learningTopics = this.mapping.filter(item => item.learner).map(item => item.topic)
+			// Filter topics based on this
+			// const topicsWithLearners = this.currentTopics.filter(item => learningTopics.includes(item.id))
+			return this.$store.state.supabase.topics
+		},
+		currentSharing: function () {
+			// Get topics that have a value in the sharing mapping 
+			// const sharingTopics = this.mapping.filter(item => item.sharer).map(item => item.topic)
+			// Filter topics based on this
+			// const topicsWithSharers = this.currentTopics.filter(item => sharingTopics.includes(item.id))
```

Comment:
similar question for these comments: still relevant?

Reply:
Removed.